### PR TITLE
[FLINK-20323][table/planner] CorrelateSortToRankRule supports dealing…

### PR DESCRIPTION
**Pull request by cshuo**
… with multiple group keys

## Brief change log

Currently, `CorrelateSortToRankRule` cannot deal with multiple group keys, e.g., 
```
  def testMultipleGroupingsNotSupported(): Unit = {
    val query =
      s"""
         |SELECT f0, f2
         |FROM
         |  (SELECT DISTINCT f0, f1 FROM t1) t2,
         |  LATERAL (
         |    SELECT f2
         |    FROM t1
         |    WHERE f0 = t2.f0 AND f1 = t2.f1
         |    ORDER BY f2
         |    DESC LIMIT 3
         |  )
      """.stripMargin
    util.verifyPlan(query)
  }
```
This pr will solve this problem.

## Verifying this change
* Unit test in `CorrelateSortToRankRuleTest`
* IT test in `RankITTest`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by cshuo** [comment]
@godfreyhe plz have a look

**Comment by cshuo** [comment]
@flinkbot run azure

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/CorrelateSortToRankRuleTest.scala [hunk 1]
```diff
@@ -95,34 +94,35 @@ class CorrelateSortToRankRuleTest extends TableTestBase {
   }
 
   @Test
-  def testAggCallNotSupported(): Unit = {
+  def testNonInnerJoinNotSupported(): Unit = {
```
Comment (tsreaper) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/CorrelateSortToRankRuleTest.scala:97]: It seems that these tests are just out of order. Reorder the tests and clean up unnecessary changes?

**Comment by tsreaper** [comment]
Also you might want to change your commit user name and email to your github user name and email.

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/CorrelateSortToRankRule.scala [hunk 2]
```diff
@@ -116,7 +121,7 @@ class CorrelateSortToRankRule extends RelOptRule(
     if (!variable.id.equals(correlate.getCorrelationId)) {
       return false
     }
-    aggInput.getInput.getDigest.equals(filter.getInput.getDigest)
+    true
```
Comment (godfreyhe) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/CorrelateSortToRankRule.scala:124]: nit: simplify these lines to `variable.id.equals(correlate.getCorrelationId)`

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/CorrelateSortToRankRuleTest.scala (hunk 1 above)
Comment (godfreyhe) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/CorrelateSortToRankRuleTest.scala:97]: +1

Comment (godfreyhe) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/CorrelateSortToRankRuleTest.scala:97]: please add some tests about equal with constant value, such as  `WHERE f0 = t2.f0 AND f1 = 1` or `WHERE f0 = t2.f0 AND 1 = t2.f1`

**Review by godfreyhe — COMMENTED** [review]
Thanks for the fix @cshuo , LGTM overall, I left some minor comments

**Review by godfreyhe — APPROVED** [review]
LGTM
